src/sentry/static/sentry/app/components/alert.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, {css} from 'react-emotion';
+import TextBlock from '../views/settings/components/text/textBlock';
+import InlineSvg from './inlineSvg';
+
+const AlertWrapper = styled.div`
+  margin: 0 0 ${p => p.theme.grid * 3}px;
+  padding: ${p => p.theme.grid * 2}px;
+  font-size: 15px;
+  box-shadow: ${p => p.theme.dropShadowLight};
+  display: flex;
+  border-radius: ${p => p.theme.borderRadius};
+
+  ${p => {
+    switch (p.type) {
```

Comment:
This switch statement is hard to read and has a lot of duplication. Instead I think we should extract this into a helper function (or functions), e.g.
https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/settings/components/panelHeader.jsx#L18

Reply:
How does that work when there are 5 possible values per property rather than two?

Reply:
Maybe we can integrate the `color` part into the `alert` object in the theme? Then we can just pass the proper alert object into the function and it will return the correct css. so something like:

```
const getAlertColorStyles = ({backgroundLight, border, iconColor}) => `
  background: ${backgroundLight};
  border: ${border}
  svg {
    color: ${iconColor};
  }
`
```

and then in the css we just do: 

`${getAlertColorStyles(p.theme.alert[p.type])}`

Reply:
^ This would be a good solution too if we had `theme.alert` structured uniformly

Reply:
Got it! I was confused because I misread and thought you were talking specifically about using the tertiary operators instead of the switch.